File: finding-seams/kotlin/src/test/kotlin/findingseams/BadgePrinterTest.kt
package findingseams

import findingseams.badgeprinting.BadgePrinting
import org.junit.jupiter.api.Disabled
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

@Disabled("Exercise: finding seams — enable when working on badge printing")
class BadgePrinterTest {
    @Test
    fun `prints a badge`() {
        val badgeDispenser = BadgePrinting()

        val badge = badgeDispenser.createBadge("Test person", "they/them")

        assertEquals("CONF-1", badge.id)
        assertEquals("Test person", badge.name)
        assertEquals("they/them", badge.pronouns)
    }

    @Test
    fun `prints second badge`() {
        val badgeDispenser = BadgePrinting()

        val badge = badgeDispenser.createBadge("Test person", "they/them")

        assertEquals("CONF-2", badge.id)
        assertEquals("Test person", badge.name)
        assertEquals("they/them", badge.pronouns)
    }
}

File: finding-seams/csharp/FindingSeams.Tests/BadgePrinterTest.cs
using FindingSeams.BadgePrinting;
using Xunit;

namespace FindingSeams.Tests;

public class BadgePrinterTest
{
    [Fact(Skip = "Exercise: finding seams — enable when working on badge printing")]
    public void Prints_a_badge()
    {
        var badgeDispenser = new BadgePrinter();

        var badge = badgeDispenser.CreateBadge("Test person", "they/them");

        Assert.Equal("CONF-1", badge.Id);
        Assert.Equal("Test person", badge.Name);
        Assert.Equal("they/them", badge.Pronouns);
    }

    [Fact(Skip = "Exercise: finding seams — enable when working on badge printing")]
    public void Prints_second_badge()
    {
        var badgeDispenser = new BadgePrinter();

        var badge = badgeDispenser.CreateBadge("Test person", "they/them");

        Assert.Equal("CONF-2", badge.Id);
        Assert.Equal("Test person", badge.Name);
        Assert.Equal("they/them", badge.Pronouns);
    }
}
